### PR TITLE
refactor(devs-infra): bring back a part of old structure

### PR DESCRIPTION
**Pull request by ahnpnl**
## Summary

- Bring back a part of old structure that `index.js`, `jest-preset.js` should stay on root dir:
    + `index.js` will point to `build/index.js`
    + `jest-preset.js` will point to `presets/index.js`

- Use `package.json` and `README` in root dir for publishing package. This will solve the issue with missing package information at https://www.npmjs.com/package/jest-preset-angular

- Create a new dir `presets`. Later if we have more presets, we can just simply modify `index.ts` to export new presets. This `index.ts` will be used by `presets` folder at root dir.

- Adjust e2e test `jest.config.js` to work with new structure.

- Use `glob` to capture which e2e test folders need to execute.

## Test plan

Green CI

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Now the `tgz` file content will be:
```
jest-preset-angular
--- build (everything from src)
--- CHANGELOG.md
--- index.js
--- jest-preset.js
--- LICENSE.md
--- package.json
--- presets
--- README.md
```

### CODE REVIEW

**Comment by ahnpnl** [comment]
I cannot find a way how to install `tgz` file for e2e tests so we still have to use `file:../../build` to install built assets. Ideally, we should test against `tgz` file so we can verify that the `tgz` file contains valid assets before publishing to npm.

The problem with `tgz` I ran into is, in my local, `tgz` after installing to e2e tests often contain stale contents, no idea how to fix that.

**Comment by ahnpnl** [comment]
Create a new entry point for `presets` will allow us to do this:
```
// jest.config.js
module.exports = {
     preset: 'jest-preset-angular/presets/foo'
}
```
with `foo` is a new preset besides the default one.

**Comment by wtho** [comment]
I do not like the overhead of files and pointers we have for `jest-preset.js` in this PR. Previously it was just one file: `src/jest-preset.js` easy to find and look into.

Now we have 4 files which look like they might contain the general content of the preset:
```
src
└── presets
     ├── create-jest-preset.ts
     └── index.ts
presets
├── index.js
└── defaults
     └── jest-preset.js
```
And this is still without the compiled files. I would prefer a single file and maybe a second one pointing at the first one.

Even for me as maintainer it will always take a while to find the right file, but imagine someone who just wants to explore the repository. This is also one of the pain points that always annoyed me in `ts-jest` when trying to find the `preset` configuration :sweat_smile: 

If we had several presets to manage, it would make sense to have a factory, although this pattern still looks like overhead to me for most scenarios.

I am not against the idea of compiling a `jest-preset.ts` to the js file, but again, a single file should be sufficient.

**Comment by ahnpnl** [comment]
done, I removed the files related to preset and only keep one `jest-preset.js` at root like originally we have

**Review by wtho — APPROVED** [review]
Thanks! Looks great!

File: .eslintignore [hunk 1]
```diff
@@ -1,7 +1,8 @@
-src/build/
-src/setup-jest.ts
+build/
 node_modules/
 e2e/
+presets/
```
[the remark this answers is not in the document]
Comment (ahnpnl) [reply .eslintignore]: done

**Pull request merged**